x-pack/legacy/plugins/lens/public/app_plugin/url_helper.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { DashboardConstants } from '../../../../../../src/legacy/core_plugins/kibana/public/dashboard';
+
+const EMPTY_DASHBOARD_PATTERN = /(.*#\/dashboard\?)(.*)/;
+const DASHBOARD_WITH_ID_PATTERN = /(.*#\/dashboard\/.*\?)(.*)/;
+const TIME_PATTERN_1 = /(.*)(,time:[^)]+\))(.*)/;
+const TIME_PATTERN_2 = /(.*)(time:[^)]+\),)(.*)/; // same as TIME_PATTERN_1, but comma follows, not preceeds
```

Comment:
Can we use `rison` to parse these instead of regexes? I think the code will be more readable.